crates/resolver/src/caching_client.rs

```diff
@@ -711,19 +783,33 @@ mod tests {
         ))
         .expect("lookup failed");
 
-        assert_eq!(
-            ips.iter().cloned().collect::<Vec<_>>(),
-            vec![
-                RData::SRV(SRV::new(
-                    1,
-                    2,
-                    443,
-                    Name::from_str("www.example.com.").unwrap(),
-                )),
-                RData::A(A::new(127, 0, 0, 1)),
-                RData::AAAA(AAAA::new(0, 0, 0, 0, 0, 0, 0, 1)),
-            ]
-        );
+        // With the new implementation, sections are preserved
```

Comment:
Nit: I think "new implementation" is not meaningful in a comment for the long term.